draft-qproto-spec.md

```diff
@@ -1,965 +1,965 @@
-as possible. As such, it uses both reliable and unreliable streams, as well
-as bidirectionality features of the transport mechanism.
-
-All data packets with descriptors `0x01**`, `0xff`, `0xfe`, `0xfd` and `0xfc`
-MUST be sent over in an *unreliable* QUIC DATAGRAM stream, as per
-IETF RFC 9221.</br>
-All other packets MUST be sent over a reliable steam. FEC data for those packets
-SHOULD NOT be signalled.
-
-Implementations MUST segment the data such that the network MTU is never
-exceeded and no packet fragmentation occurs.</br>
-The **minimum** network MTU required for the protocol is **384 bytes**,
-as to allow [video information packets](#video-info-packets) or any future large
-packets to be sent without fragmentation.
-
-Jumbograms MAY be used where supported to reduce overhead and increase efficiency.
-
-Reverse signalling
-==================
-The following section is a specification on how reverse signalling, where
-receiver(s) communicates with the sender, should be formatted.
-
-The same port an method MUST be used for reverse signalling as the sender's.
-
-Control data
-------------
-The receiver can use this type to return errors and more to the sender in a
-one-to-one transmission. The following syntax is used:
-
-| Data                | Name               | Fixed value  | Description                                                                                              |
-|:--------------------|:-------------------|-------------:|:---------------------------------------------------------------------------------------------------------|
-| b(16)               | `ctrl_descriptor`  |       0x8001 | Indicates this is a control data packet.                                                                 |
-| b(8)                | `cease`            |              | If not equal to `0x0`, indicates a fatal error, and senders MUST NOT sent any more data.                 |
-| u(32)               | `error`            |              | Indicates an error code, if not equal to `0x0`.                                                          |
-| b(8)                | `resend_init`      |              | If nonzero, asks the sender to resend `time_sync` and all stream `init_data` packets.                    |
-| b(128)              | `uplink_ip`        |              | Reports the upstream address to stream to.                                                               |
-| b(16)               | `uplink_port`      |              | Reports the upstream port to stream on to the `uplink_ip`.                                               |
-| i(64)               | `seek`             |              | Asks the sender to seek to a specific relative byte offset, as given by [index_packets](#index-packets). |
-
-If the sender gets such a packet, and either its `uplink_ip` or its `uplink_port`
-do not match, the sender **MUST** cease this connection, reopen a new connection
```

Comment:
This assumes secure transport or authenticated packets, such that all packets are guaranteed to originate from the authentic sender. Otherwise, anyone can just send malicious packets and the spec would require dropping the connection. (Or did you account for this situation already?)

If this section is subject to chapter on QUIC, then I misread and this comment is mute.